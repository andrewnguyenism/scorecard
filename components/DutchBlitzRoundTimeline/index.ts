export { DutchBlitzRoundTimeline } from './DutchBlitzRoundTimeline'

export type { DutchBlitzRound } from './DutchBlitzRoundTimeline'
