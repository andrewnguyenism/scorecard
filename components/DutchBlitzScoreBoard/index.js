export {
  DutchBlitzScoreBoardHeader,
  DutchBlitzScoreBoardPlayer,
  DutchBlitzScoreBoard,
} from "./DutchBlitzScoreBoard";
