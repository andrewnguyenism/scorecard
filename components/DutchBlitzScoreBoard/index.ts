export { DutchBlitzScoreBoard } from './DutchBlitzScoreBoard'
export { DutchBlitzScoreBoardHeader } from './DutchBlitzScoreBoardHeader'
export { DutchBlitzScoreBoardPlayer } from './DutchBlitzScoreBoardPlayer'
