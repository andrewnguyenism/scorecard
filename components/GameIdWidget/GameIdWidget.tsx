import { createRef, FunctionComponent } from "react";

interface Props {
  gameId: string;
}

export const GameIdWidget: FunctionComponent<Props> = ({ gameId }) => {
  const ref = createRef<HTMLDivElement>();

  const copyToClipboard = () => {
    navigator.clipboard.writeText(gameId);
  };

  return (
    <div
      className="cursor-pointer rounded-2xl bg-gray-300 hover:bg-gray-400 px-3 inline-flex items-center font-bold"
      onClick={copyToClipboard}
      ref={ref}
      role="button"
    >
      <span className="text-gray-800 text-xl font-mono font-bold tracking-widest">
        {gameId}
      </span>
      <svg
        className="h-4 w-4 ml-0.5"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 20 20"
        fill="currentColor"
      >
        <path d="M8 2a1 1 0 000 2h2a1 1 0 100-2H8z" />
        <path d="M3 5a2 2 0 012-2 3 3 0 003 3h2a3 3 0 003-3 2 2 0 012 2v6h-4.586l1.293-1.293a1 1 0 00-1.414-1.414l-3 3a1 1 0 000 1.414l3 3a1 1 0 001.414-1.414L10.414 13H15v3a2 2 0 01-2 2H5a2 2 0 01-2-2V5zM15 11h2a1 1 0 110 2h-2v-2z" />
      </svg>
    </div>
  );
};
