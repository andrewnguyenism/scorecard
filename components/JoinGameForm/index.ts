export { JoinGameFormError, JoinGameForm } from './JoinGameForm'
