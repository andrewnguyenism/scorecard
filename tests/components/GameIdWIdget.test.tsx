import { fireEvent, render, screen } from '../test-utils'

import { GameIdWidget } from '@/components/GameIdWidget'

Object.assign(navigator, {
  clipboard: {
    writeText: jest.fn()
  }
})

describe('GameIdWidget component', () => {
  it('renders component', () => {
    const { asFragment } = render(<GameIdWidget gameId="AB123" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('copies to clipboard when clicked', () => {
    const gameId = 'AB123'
    render(<GameIdWidget gameId={gameId} />)

    fireEvent.click(screen.getByText(gameId))

    expect(navigator.clipboard.writeText).toHaveBeenCalledWith(gameId)
  })
})
