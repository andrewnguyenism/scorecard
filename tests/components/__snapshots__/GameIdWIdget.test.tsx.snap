// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameIdWidget component renders component 1`] = `
<DocumentFragment>
  <div
    class="cursor-pointer rounded-2xl bg-gray-300 hover:bg-gray-400 px-3 inline-flex items-center font-bold"
    role="button"
  >
    <span
      class="text-gray-800 text-xl font-mono font-bold tracking-widest"
    >
      AB123
    </span>
    <svg
      class="h-4 w-4 ml-0.5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 2a1 1 0 000 2h2a1 1 0 100-2H8z"
      />
      <path
        d="M3 5a2 2 0 012-2 3 3 0 003 3h2a3 3 0 003-3 2 2 0 012 2v6h-4.586l1.293-1.293a1 1 0 00-1.414-1.414l-3 3a1 1 0 000 1.414l3 3a1 1 0 001.414-1.414L10.414 13H15v3a2 2 0 01-2 2H5a2 2 0 01-2-2V5zM15 11h2a1 1 0 110 2h-2v-2z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
